src/components/vm/disks/diskAdd.jsx

```diff
@@ -513,11 +515,41 @@ export class AddDiskModalBody extends React.Component {
             break;
         }
         case 'file': {
+            this.setState({ file: value });
+
             if (value.endsWith(".iso")) {
                 // use onValueChange instead of setState in order to perform subsequent state change logic
                 this.onValueChanged("device", "cdrom");
             }
-            this.setState({ file: value });
+
+            if (value) {
+                cockpit.spawn(["head", "--bytes=16", value], { binary: true, err: "message", superuser: "try" })
+                        .then(output => {
+                            let format = "";
+                            const file_header = Array.from(output);
```

Comment:
What does this do? `output` is already an array, so this smells like a no-op? I.e. you could just directly do `.then(file_header => ...)` instead?

Reply:
I think it's an UIntArray which is special, let me second check.

Reply:
Turns out, I'm incorrect, we can slice!